nilearn/glm/tests/test_second_level.py

```diff
@@ -1384,3 +1394,148 @@ def test_second_lvl_dataframe_computation(tmp_path, shape_3d_default):
+
+
+# -----------------------surface tests----------------------- #
+
+
+def test_second_level_input_as_surface_image(surf_img):
+    """Test slm with a list surface images as input."""
+    n_subjects = 5
+    second_level_input = [surf_img() for _ in range(n_subjects)]
+
+    design_matrix = pd.DataFrame(
+        [1] * len(second_level_input), columns=["intercept"]
+    )
+
+    model = SecondLevelModel()
+    model = model.fit(second_level_input, design_matrix=design_matrix)
+
+
+def test_second_level_input_as_surface_image_3d(surf_img):
+    """Fit with surface image with all subjects as timepoints."""
+    n_subjects = 5
+    second_level_input = surf_img(n_subjects)
+
+    design_matrix = pd.DataFrame([1] * n_subjects, columns=["intercept"])
+
+    model = SecondLevelModel()
+
+    model.fit(second_level_input, design_matrix=design_matrix)
+
+
+def test_second_level_input_as_surface_image_3d_same_as_list_2d(surf_img):
+    """Fit all subjects as timepoints same as list of subject."""
+    n_subjects = 5
+    second_level_input = [surf_img() for _ in range(n_subjects)]
+
+    design_matrix = pd.DataFrame([1] * n_subjects, columns=["intercept"])
+
+    model = SecondLevelModel()
+    model.fit(second_level_input, design_matrix=design_matrix)
+    result_2d = model.compute_contrast()
+
+    second_level_input_3d = concatenate_surface_images(second_level_input)
+    model.fit(second_level_input_3d, design_matrix=design_matrix)
+    result_3d = model.compute_contrast()
+
+    assert_surface_image_equal(result_2d, result_3d)
+
+
+def test_second_level_input_as_surface_no_design_matrix(surf_img):
+    """Raise error when design matrix is missing."""
+    n_subjects = 5
+    second_level_input = [surf_img() for _ in range(n_subjects)]
+
+    model = SecondLevelModel()
+
+    with pytest.raises(
+        ValueError, match="require a design matrix to be provided"
+    ):
+        model.fit(second_level_input, design_matrix=None)
+
+
+def test_second_level_input_as_surface_image_with_mask(surf_img, surf_mask):
+    """Test slm with surface mask and a list surface images as input."""
+    n_subjects = 5
+    second_level_input = [surf_img() for _ in range(n_subjects)]
+
+    design_matrix = pd.DataFrame(
+        [1] * len(second_level_input), columns=["intercept"]
+    )
+
+    model = SecondLevelModel(mask_img=surf_mask())
+    model = model.fit(second_level_input, design_matrix=design_matrix)
+
+
+def test_second_level_input_as_surface_image_warning_smoothing(surf_img):
+    """Test slm with a list surface images as input."""
+    n_subjects = 5
+    second_level_input = [surf_img() for _ in range(n_subjects)]
+
+    design_matrix = pd.DataFrame(
+        [1] * len(second_level_input), columns=["intercept"]
+    )
+
+    model = SecondLevelModel(smoothing_fwhm=8.0)
+    # TODO
+    # should throw warning that smoothing is not implemented
+    model = model.fit(second_level_input, design_matrix=design_matrix)
```

Comment:
Maybe a check for the warning is needed here?

Reply:
good catch!!
thanks

Reply:
fixed